src/compiler/moduleNameResolver.ts

```diff
@@ -343,7 +343,7 @@ namespace ts {
                     trace(host, Diagnostics.Resolving_with_primary_search_path_0, typeRoots.join(", "));
                 }
                 return firstDefined(typeRoots, typeRoot => {
-                    const candidate = combinePaths(typeRoot, typeReferenceDirectiveName);
+                    const candidate = combinePaths(typeRoot, mangleScopedPackageNameWithTrace(typeReferenceDirectiveName, moduleResolutionState));

```

Comment:
If this is not from `node_modules` are we sure we want to look for mangled type name? Not sure what the rules are. @RyanCavanaugh @DanielRosenwasser to chime in.

Reply:
Apologies, I think the original PR name was misleadingly broad - hopefully this is more representative of the actual change.

This primary lookup (from what I can see) is searching through the [`typeRoots`](https://www.typescriptlang.org/tsconfig#typeRoots) configured in the compiler options (or ~`node_modules/@types` by default)

The non-`node_modules` lookups appear to happen within the secondary lookup and in that case the mangling is suitably guarded and will only occur when looking in an `@types` subdirectory ([here](https://github.com/microsoft/TypeScript/blob/19db9ad6757858a117b3c1c2cc9ad75ad7537e8f/src/compiler/moduleNameResolver.ts#L1295-L1303))

If I understood the above correctly (big if! :sweat_smile:), given that this primary lookup is specifically looking within `typeRoots` directories, I feel like it's reasonable to also support mangling there too